### PR TITLE
Added lifestone Materialize, Movement and Position Update Event

### PR DESCRIPTION
**Pull request by fantoms**
Please check to verify that I made the correct assumptions on this one. Documentation used: 

http://skunkworks.sourceforge.net/protocol/Protocol.php?stylesheet=Classic.css&frame=frameMsg&type=F619

http://www.virindi.net/junk/messages_annotated_final.xml :

`<message type="F619" name="Lifestone Materialize" text="Sent when a character rematerializes at the lifestone after death." lastUpdate="20050918" lastUpdater="kgober"><field type="ObjectID" name="object" text="ObjectID of the character doing the animation"/><field type="Position" name="position"/><field type="DWORD" name="unknown2" text="Unknown"/><field type="DWORD" name="unknown3" text="Unknown"/><field type="DWORD" name="unknown4" text="Unknown"/><field type="DWORD" name="unknown5" text="Unknown"/></message>`

### CODE REVIEW

File: aclogview/CM_Movement.cs [hunk 1]
```diff
@@ -664,6 +669,44 @@ public class Jump_NonAutonomous : Message {
...
+        public uint unknown2;
+        public uint unknown3;
+        public uint unknown4;
+        public uint unknown5;
+
+        public static LifestoneMaterialize read(BinaryReader binaryReader)
+        {
+            LifestoneMaterialize newObj = new LifestoneMaterialize();
+            newObj.ObjectId = binaryReader.ReadInt32();
+            newObj.Position = Position.read(binaryReader);
```
Comment (tfarley) [inline aclogview/CM_Movement.cs]: Looks like there are 4 unknown bytes between the object id and the position. It also looks like there are at least 24 bytes following the position. Suggest adding an unknown int between id and position as well as adding two more unknown ints to the end to fully read the packet.

File: aclogview/CM_Movement.cs [hunk 2]
```diff
@@ -664,6 +669,44 @@ public class Jump_NonAutonomous : Message {
...
+        public Position Position;
+        public uint unknown2;
+        public uint unknown3;
+        public uint unknown4;
+        public uint unknown5;
+
+        public static LifestoneMaterialize read(BinaryReader binaryReader)
+        {
+            LifestoneMaterialize newObj = new LifestoneMaterialize();
+            newObj.ObjectId = binaryReader.ReadInt32();
```
Comment (tfarley) [inline aclogview/CM_Movement.cs]: Id should be unsigned int.

**Review by tfarley — CHANGES_REQUESTED** [review]

**Comment by fantoms** [comment]
Thank you!

I've updated this branch with your suggestions; please review and let me know what you think. I'm seeing the results of "WARNING: Prev fragment not fully read!".

File: aclogview/CM_Movement.cs [hunk 3]
```diff
@@ -664,6 +669,40 @@ public class Jump_NonAutonomous : Message {
...
+        public int unknown3;
+
+        public static LifestoneMaterialize read(BinaryReader binaryReader)
+        {
+            LifestoneMaterialize newObj = new LifestoneMaterialize();
+            newObj.ObjectId = binaryReader.ReadUInt32();
+            newObj.unknown1 = binaryReader.ReadInt32();
+            newObj.Position = Position.read(binaryReader);
+            newObj.unknown2 = binaryReader.ReadInt32();
+            newObj.unknown3 = binaryReader.ReadInt32();
```
Comment (tfarley) [inline aclogview/CM_Movement.cs:691]: Thanks for the changes. Was intending for two _more_ ints after position though - id, unk1, pos, unk2, unk3, unk4, unk5, unk6, unk7. Naturally we have no idea if those are actually ints, but at least the whole fragment gets read that way and you won't get that warning.

**Review by tfarley — CHANGES_REQUESTED** [review]

File: aclogview/CM_Movement.cs (hunk 3 above)
Comment (fantoms) [reply aclogview/CM_Movement.cs:691]: Perfect, I was also getting quite confused as to how that would work. Thank you for the further clarification, it should now be set!

**Comment by fantoms** [comment]
Should be ready to review, the warning has now disappeared.

**Comment by tfarley** [comment]
Great, thanks. Sorry for the confusion. :)

**Pull request merged**